tests/test-pkcs11-tool-sign-verify.sh

```diff
@@ -205,34 +207,44 @@ echo "Test ECDSA keys"
 echo "======================================================="
 # operations with ECDSA keys should work on data > 512 bytes; generate data:
 head -c 1024 </dev/urandom > data
-for SIGN_KEY in "03" "04"; do
-    METHOD="ECDSA"
+for HASH in "SHA1" "SHA256"; do
+    for SIGN_KEY in "03" "04"; do
+        METHOD="ECDSA"
 
-    echo
-    echo "======================================================="
-    echo "$METHOD: Sign & Verify (KEY $SIGN_KEY)"
-    echo "======================================================="
-    openssl dgst -binary -sha256 data > data.hash
-    $PKCS11_TOOL "${PRIV_ARGS[@]}" --id $SIGN_KEY -s -m $METHOD \
-        --input-file data.hash --output-file data.sig
-    assert $? "Failed to Sign data"
-    $PKCS11_TOOL "${PRIV_ARGS[@]}" --id $SIGN_KEY -s -m $METHOD \
-        --input-file data.hash --output-file data.sig.openssl \
-        --signature-format openssl
-    assert $? "Failed to Sign data into OpenSSL format"
+        echo
+        echo "======================================================="
+        echo "$METHOD+$HASH: Sign & Verify (KEY $SIGN_KEY)"
+        echo "======================================================="
+        
+        # Skip SHA-1 operations only in wolfProvider FIPS mode
+        if [[ "${ENABLE_WOLFPROV}" = "1" && "${WOLFSSL_ISFIPS}" = "1" && "$HASH" = "SHA1" ]]; then
```

Comment:
Is there some way to detect the FIPS mode in some more generic way? Currently we have FIPS mode detection already in here, which is how it works in RHEL-based OS:

https://github.com/OpenSC/OpenSC/blob/master/tests/test-p11test.sh#L43

If we are going to add more conditions, I would prefer to have the check moved into the `common.sh`, rather than scattered through the files and in tests, just check `FIPS_MODE` variable for better readability. The SHA1 will not be enabled in other FIPS providers either.